lib/src/impl/rest/rest.dart

```diff
@@ -7,6 +7,7 @@ import '../../spec/spec.dart' as spec;
 import '../platform_object.dart';
 import 'channels.dart';
 
+Map<int, Rest> restInstances = {};
```

Comment:
@zoechi restInstances is accessed in `AblyMethodCallHandler#onAuthCallback` (method_call_handler.dart)

We can alternatively have a static method on `Rest` like `Rest.getInstanceByHandle(int handle)` to get a rest instance and make `restInstances` as private.

Reply:
It's not part of this PR but at line 38
```dart
@override
ClientOptions options;
```
probably should be
```dart
@override
final ClientOptions options;
```
otherwise the asserts in the constructor do not make sense.

For this you probably need to change `spec/rest/ably_base.dart`
```dart
  ClientOptions options;
``` 
to
```dart
  ClientOptions get options;
```
Please also check all other fields in `ably_base.dart` if they should actually provide a setter as well, or if just getters are enough for these interfaces.

Reply:
I assume that accesses from outside this file should only be allowed to read, but not to add, remove, or replace?
In this case I'd use
```dart
import 'dart:collection';
...
Map<int, Rest> _restInstances = {};
Map<int, Rest> _restInstancesUnmodifiableView;
Map<int, Rest> get restInstances => _restInstancesUnmodifiableView ??= UnmodifiableMapView(_restInstances);
```
and change update access in the rest of the file to the private `_restInstances`

A bit verbose, but it communicates clearly that modifications are not supposed to be made.
`UnmodifiableMapView` does not provide feedback through static analyis, it only throws at modification attempts. The advantage is that it passes as regular `Map`.

Reply:
@zoechi 
About your previous comment:
> It's not part of this PR but at line 38...

These values can be changed any time, they need not be locked on initializing. At least that's what I could infer from [the IDL](https://docs.ably.io/client-lib-development-guide/features/#idl)

There are many other classes with such behavior in `spec/` folder. Can we have an issue to discuss this?

Reply:
25bda8181091fce8e7b86b12804d26e55ce5e581

Reply:
@zoechi can you resolve this conversation if you're happy with @tiholic's change?

Reply:
Sorry, I missed these comments. 
Why did you change it when they are supposed to be changeable?
Turned it out that they actually shouldn't be modifiable?